### DISCUSSION TITLE
Snapshot URL that serves actual snapshot

### DISCUSSION THREAD

**Discussion by bwucke** [Motion-Project/motion#1628] [General]
url:8081/ serves the live stream from the camera
url:8080/0/action/snapshot  creates the snapshot - on disk. On www, in the http reply, you get the message that a snapshot for camera 0 is done. Go get it through ftp or something if you want to see it.

Would it be a big problem to give Motion an URL that creates a snapshot which it serves as reply for the request?  Say, url:8081/snapshot which would return a single still jpg captured at the moment of calling it the way url:8080 returns a live video?

Motion is one of best choice streaming servers for 3D printing with Octoprint - not only does it run fine with cameras on which mjpgstreamer croaks, it's also much easier to set-up and feature-rich. And it allows monitoring of print progress/problems just fine, but you can't use it for timelapses which take snapshots on input. People overcome the problem through assortment of scripts running on webservers they set up just for the purpose of serving the snapshots, but it would be very nice if it was supported natively in Motion.

**Comment by Mr-Dave** [comment]
I'm not sure I'm following so let me explain what can be done and you can fill in the missing bit.....For version 4.5.1 you can specify a URL such as `http://10.0.0.10:8081/84/current` and it will return a single jpg image.  I suppose you could also use wget and have it save to a file similar to the snapshot functionality... Additionally Motion can use that same URL as an input.  (Make sure to use 4.5.1 or manually specify all the required additional ffmpeg parameters).  Motion can also be setup to create a timelapse of that input based upon the `timelapse` parameters.  What is missing?
